rc/tools/git.kak

```diff
@@ -42,6 +42,12 @@ hook -group git-show-branch-highlight global WinSetOption filetype=git-show-bran
 declare-option -hidden line-specs git_blame_flags
 declare-option -hidden line-specs git_diff_flags
 declare-option -hidden int-list git_hunk_list
+declare-option -hidden \
```

Comment:
@krobelus Maybe this option should not even be hidden. It's quite useful for scripting.

Very nice command that I have in my private config (which could possibly be a useful addition to the build-in kakscripts) is `git show` of the commit that the given line blames.

```
define-command git-show-blamed -params .. \
    -docstring %{
        git-show-blamed [<arguments>]: show the commit that is blamed for the current line
        All arguments are forwarded to git blame
    } \
%{
    git show %sh{
        # Change directory to where the relevant repo is supposed to be
        cd $kak_opt_git_work_tree_top_level
        git blame \
            "$kak_buffile" \
            -L"$kak_cursor_line,$kak_cursor_line" \
            --porcelain $@ \
            | head -1 | cut -d' ' -f1
    }
}
```

Reply:
the command looks nice. Should quote `'$@"` to be safe.
I don't think it requires your option; we  can cd to the buffile directory instead.
I personally use `tig` because it supports recursive blame.
Actually it seems doable to implement recursive-blame support in your command.

Related question is whether we should have a third-party plugin for Git. Maybe that would enable experiments with various interfaces.
I imagine that after a `git-blame`, `<ret>` should do what your command does. I guess this is something that we could try to add to the built-in scripts.. `<ret>` is currently only used by *grep* and `git-diff-goto-source`.. the latter would collide with recursive blame.. not sure how to fix that. Could take a look at  similar plugins

Reply:
oh I see, `*git*` does not have an implied git directory that's why the option is useful here.
Given that, I'm happy to champion a `git_work_tree` option that overrides $PWD* as long as this option is only set in the buffer scope

* (maybe also overrides the buffer directory but that doesn't really matter for scratch buffers)

Reply:
> I personally use tig because it supports recursive blame.

I use `gitui` usually, that one does not seem to have it. Being able to recursively traverse the blame history would be really neat indeed although..

> Actually it seems doable to implement recursive-blame support in your command.

> the latter would collide with recursive blame

Yeah, not sure how that would work either 🤔 That sounds like a slightly more advance assignment. One could checkout a single file from the commit pointed by the blame into the scratch buffer and apply `run_git_blame` with the `git blame` output from that commit perspective. And that should work recursively until I get to the first commit that given line was introduce at.